src/Traits/SortableTrait.php

```diff
@@ -2,18 +2,54 @@
 
 namespace FluxErp\Traits;
 
+use FluxErp\Models\Scopes\OrderedScope;
+use Illuminate\Database\Eloquent\Model;
 use Spatie\EloquentSortable\SortableTrait as BaseSortableTrait;
 
 trait SortableTrait
 {
-    use BaseSortableTrait;
+    private bool $isSorted = false;
+
+    use BaseSortableTrait {
+        BaseSortableTrait::bootSortableTrait as bootParentSortableTrait;
+    }
+
+    protected static function bootSortableTrait(): void
+    {
+        static::addGlobalScope(OrderedScope::class);
+
+        static::bootParentSortableTrait();
+
+        static::saving(function (Model $model) {
+            $orderColumn = $model->determineOrderColumnName();
+
+            if ($model->isDirty($orderColumn) && ! $model->getIsSorted()) {
+                $newPosition = $model->$orderColumn;
+                $model->$orderColumn = $model->getRawOriginal($orderColumn);
+
+                $model->moveToPosition($newPosition);
+            }
+        });
+    }
+
+    public function setIsSorted(bool $isSorted = true): static
```

Comment:
must this be public or can this be protected? Just curious.

Reply:
can be protected, changed